src/components/Widgets/NumberControl.js

```diff
@@ -1,17 +1,34 @@
 import React, { PropTypes } from 'react';
 
-export default class StringControl extends React.Component {
+export default class NumberControl extends React.Component {
   handleChange = (e) => {
-    this.props.onChange(e.target.value);
+    const valueType = this.props.field.get('valueType')
+    let value
```

Comment:
We generally want to avoid mutating variables - you could instead set `onChange` as a new `const` and call it in each condition, without storing the value.